src/Pos/Block/Network/Retrieval.hs

```diff
@@ -135,9 +135,8 @@ dropRecoveryHeader peerId = do
 dropRecoveryHeaderAndRepeat
     :: (SscWorkersClass ssc, WorkMode ssc m)
     => SendActions m -> NodeId -> m ()
-dropRecoveryHeaderAndRepeat sendActions peerId = do
-    kicked <- dropRecoveryHeader peerId
-    when kicked $ attemptRestartRecovery
+dropRecoveryHeaderAndRepeat sendActions peerId =
+    whenM (dropRecoveryHeader peerId) attemptRestartRecovery
```

Comment:
This is actually one of the cases where it would've been better to leave `kicked <-` and avoid applying the suggestion (such cases are the reason why I made `whenM` a suggestion and not a warning)